sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/parallel_for.py

```diff
@@ -0,0 +1,170 @@
+        return cls._create_instance_from_schema_dict(pipeline_jobs=pipeline_jobs, loaded_data=obj)
+
+    @classmethod
+    def _create_instance_from_schema_dict(cls, pipeline_jobs, loaded_data):
+        body_name = cls._get_data_binding_expression_value(loaded_data.pop("body"), regex=r"\{\{.*\.jobs\.(.*)\}\}")
+
+        loaded_data["body"] = cls._get_body_from_pipeline_jobs(pipeline_jobs=pipeline_jobs, body_name=body_name)
+        return cls(
+            **loaded_data
+        )
+
+    def _validate_items(self, raise_error=True):
+        validation_result = self._create_empty_validation_result()
+        if self.items is not None:
+            items = self.items
+            if isinstance(items, str):
+                # try to deserialize str if it's a json string
+                try:
+                    items = json.loads(items)
+                except json.JSONDecodeError as e:
+                    if not is_data_binding_expression(items, ["parent"]):
+                        validation_result.append_error(
+                            f"Items is neither a valid JSON string due to {e} or a binding string."
+                        )
+            if isinstance(items, dict):
+                # Validate dict keys
+                items = list(items.values())
+            if isinstance(items, list):
+                if len(items) > 0:
+                    self._validate_items_list(items, validation_result)
+                else:
+                    validation_result.append_error("Items is an empty list/dict.")
+        else:
+            validation_result.append_error(
+                "Items is required for parallel_for node",
+            )
+        return validation_result.try_raise(self._get_validation_error_target(), raise_error=raise_error)
+
+    def _customized_validate(self):
+        """Customized validation for parallel for node."""
+        validation_result = self._validate_body(raise_error=False)
+        validation_result.merge_with(self._validate_items(raise_error=False))
+        return validation_result
+
+    def _validate_items_list(self, items: list, validation_result):
+        # pylint: disable=protected-access
+        meta = {}
+        # all items have to be dict and have matched meta
+        for item in items:
+            # item has to be dict
+            if not isinstance(item, dict) or item == {}:
+                validation_result.append_error(
+                    f"Items has to be list/dict of non-empty dict as value, "
+                    f"but got {type(item)} for {item}."
+                )
+            else:
+                # item has to have matched meta
+                if meta.keys() != item.keys():
+                    if not meta.keys():
+                        meta = item
+                    else:
+                        validation_result.append_error(
+                            f"Items should to have same keys, but got {item.keys()} and {meta.keys()}."
```

Comment:
Items should to have same keys with body inputs

Reply:
Updated